BitcoinApp/CoinViewController/CoinsViewController.swift

```diff
@@ -0,0 +1,101 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        guard let currency = Currency(rawValue: currencySegmentControl.selectedSegmentIndex) else { return }
+        guard  let period = Period(rawValue: periodSegmentControl.selectedSegmentIndex) else { return }
+        
+        configureChartView()
+        configureSegmentControl()
+        updateCurrencyValue(currency: currency)
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    //MARK: - @IBActions
+    
+    @IBAction func currencySegmentControlSelect(_ sender: UISegmentedControl) {
+        guard let currency = Currency(rawValue: sender.selectedSegmentIndex) else { return }
+        updateCurrencyValue(currency: currency)
+        guard let period = Period(rawValue: periodSegmentControl.selectedSegmentIndex) else { return }
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    @IBAction func periotSegmentControlSelect(_ sender: UISegmentedControl) {
+        guard let period = Period(rawValue: sender.selectedSegmentIndex) else { return }
+        guard let currency = Currency(rawValue: currencySegmentControl.selectedSegmentIndex) else { return }
+        updateCurrencyHistory(currency: currency, period: period)
+    }
+    
+    //MARK: - getData
+    
+    func updateCurrencyHistory(currency: Currency, period: Period) {
+        HistoryNetworkService.getHistory(currency: currency, period: period) { (dataY) in
+            DispatchQueue.main.async {
+                self.setChart(values: dataY, currency: currency, period: period)
+            }
+        }
+    }
+    
+    func updateCurrencyValue(currency: Currency) {
+        CoinNetworkService.getCoin(currency: currency) { (coin) in
+            DispatchQueue.main.async {
+                self.rateValueLabel.text = "\(coin.rateFloat.formatingToString())\(currency.text)"
+            }
+        }
+    }
+    
+    //MARK: - Configure
+    
+    func configureSegmentControl() {
+        periodSegmentControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.white], for: .normal)
+        currencySegmentControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.white], for: .normal)
+    }
+    
+    func configureChartView() {
+        bitcoinChartView.xAxis.drawLabelsEnabled = false
+        bitcoinChartView.leftAxis.enabled = false
+        bitcoinChartView.leftAxis.drawGridLinesEnabled = false
+        bitcoinChartView.rightAxis.enabled = false
+        bitcoinChartView.rightAxis.drawGridLinesEnabled = false
+        bitcoinChartView.backgroundColor = UIColor(red: 10/255, green: 10/255, blue: 10/255, alpha: 1)
+    }
+    
+    func setChart (values: [Double], currency: Currency, period: Period) {
+        var yValues = [ChartDataEntry]()
+        for i in 0..<values.count {
+            let dataEntry = ChartDataEntry(x: Double(i + 1), y: values[i])
```

Comment:
There is a safer call in Swift Array: `values.enumerated`. Your for-in implementation would get two arguments - the index at first, and the object as a second.
It would look like this:
`for (index, value) in values.enumerated()`